exporter/collector/integrationtest/testcases/conversion.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testcases
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// ConvertResourceMetrics converts a (collector) pdata metrics to an SDK ResourceMetrics
+// This is useful for testing the SDK with the same data used to test the collector.
+func ConvertResourceMetrics(pdataMetrics pmetric.Metrics) []metricdata.ResourceMetrics {
```

Comment:
the conversion all looks pretty 1:1, but are you worried at all about maintaining these conversions? The metrics sdk is only alpha, so it could still change right?

Reply:
It is, but it is spec-compliant (and the spec is stable), and the underlying OTLP datamodel is also stable, so I don't think it is likely to change.  pdata APIs are also not stable, so its also a risk.  But i'm not too worried.